### PR TITLE
Add a modal window when the user clicks on 'Regenerate thumbnails' button

### LINKED ISSUE #20962: By default, turn off the 'Erase previous images' button when regenerating thumbnails
#### Is your feature request related to a problem

Accidentaly deleted all the images trying to regenerate for the new products.

#### Describe the solution you'd like - Edited by Louise B. on September 28th following https://github.com/PrestaShop/PrestaShop/issues/20962#issuecomment-692632521:

I would like to propose, the default state of "delete all images" to be off. Accidents can happen when regenerating for new products. 

- [x] 1. Emphasize the irreversibility of this action by **turning the blue information notification into a yellow warning one**, stating: / Fixed by #21215

_Be careful! Depending on the options selected, former manually uploaded thumbnails might be erased and replaced by automatically generated thumbnails. / Also, regenerating thumbnails for all existing images can take several minutes, please be patient._ in _Admin.Design.Notification_. 

- [x] 2. In order to avoid previous work to be deleted by mistake, **set the 'Erase previous images' switch button to 'No'** by default. / Fixed by #21166

- [x] 3. Update the modal window that pops up when the user clicks on 'Regenerate thumbnails' to **display a more explicit message**: / Fixed by #30054 

`title` = _Regenerate thumbnails_ in _Admin.Actions_
`confirm_message` = _Are you sure you want to regenerate thumbnails for the selected images? With the erase option enabled, the previous thumbnails of your selection will be deleted._ in _Admin.Design.Notification_

`confirm_button_label` = _Regenerate_ in _Admin.Actions_
`close_button_label` = _Cancel_ in _Admin.Actions_

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add a modal window when the user clicks on 'Regenerate thumbnails' button
| Type?             | new feature
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #20962
| Related PRs       | N/A
| How to test?      | In Image Settings, a modal window has been added. The wording come from https://github.com/PrestaShop/PrestaShop/issues/20962#issue-699896621
| Possible impacts? | N/A

### CODE REVIEW

**Review by matks — DISMISSED** [review]

File: admin-dev/themes/default/template/controllers/images/content.tpl
Comment (Hlavtox) [inline admin-dev/themes/default/template/controllers/images/content.tpl]:
```html
<button 
type="submit" 
name="submitRegenerate{$table}" 
value="" 
id="submitRegenerate{$table}" 
class="btn btn-default pull-right"
>
```

**Review by Hlavtox — DISMISSED** [review]

File: admin-dev/themes/default/template/controllers/images/content.tpl
Comment (NeOMakinG) [inline admin-dev/themes/default/template/controllers/images/content.tpl]: Why didn't you choose to implement it inside a JS file directly? Is there a valid reason?

**Comment by nicosomb** [comment]
@Progi1984 do we need to link your PR with https://github.com/PrestaShop/PrestaShop/issues/27318?

File: admin-dev/themes/default/template/controllers/images/modal_regenerate_thumbnails.tpl
Comment (l-delin) [inline admin-dev/themes/default/template/controllers/images/modal_regenerate_thumbnails.tpl:27]:
```suggestion
    s="Are you sure you want to regenerate thumbnails for the selected images? With the erase option enabled, the previous thumbnails that you selected will be deleted."
```

Comment (l-delin) [inline admin-dev/themes/default/template/controllers/images/modal_regenerate_thumbnails.tpl]:
```suggestion
    s="Regenerate thumbnails for the selected images? With the erase option enabled, the previous thumbnails that you selected will be deleted."
```

Comment (Progi1984) [reply admin-dev/themes/default/template/controllers/images/modal_regenerate_thumbnails.tpl]: Fixed

File: admin-dev/themes/default/template/controllers/images/content.tpl
Comment (Progi1984) [reply admin-dev/themes/default/template/controllers/images/content.tpl]: Fixed

File: admin-dev/themes/default/template/controllers/images/content.tpl
Comment (Progi1984) [reply admin-dev/themes/default/template/controllers/images/content.tpl]: Yes, it's legacy on the first part, and I use a smarty variable.

**Comment by Progi1984** [comment]
@PrestaShop/prestashop-maintainers  / @PrestaShop/committers : Fixed feedbacks

@l-delin : Fixed wording

Thanks for your time & your review.

**Review by nicosomb — DISMISSED** [review]

**Review by zuk3975 — DISMISSED** [review]

File: admin-dev/themes/default/template/controllers/images/content.tpl
Comment (Hlavtox) [reply admin-dev/themes/default/template/controllers/images/content.tpl]: @Progi1984 You can remove the `<input type="hidden" name="submitRegenerate{$table}" value="" />`, it should work the same. 😎

Comment (Progi1984) [reply admin-dev/themes/default/template/controllers/images/content.tpl]: Just reverted your fix. Because the modal doesn't POST the button data, so it was the cause I implemented an hidden input.

**Comment by Progi1984** [comment]
@zuk3975 / @nicosomb Sorry : I rebased the PR.

@l-delin Could you review the wording ? Thanks.

**Review by zuk3975 — DISMISSED** [review]

File: controllers/admin/AdminImagesController.php
Comment (l-delin) [inline controllers/admin/AdminImagesController.php]:
```suggestion
            'modal_title' => $this->trans('Regenerate thumbnails', [], 'Admin.Design.Feature'),
```

File: controllers/admin/AdminImagesController.php
Comment (l-delin) [inline controllers/admin/AdminImagesController.php]:
```suggestion
                    'label' => $this->trans('Regenerate', [], 'Admin.Design.Feature'),
```

**Comment by l-delin** [comment]
> @zuk3975 / @nicosomb Sorry : I rebased the PR.
> 
> @l-delin Could you review the wording ? Thanks.

done 🙌

File: controllers/admin/AdminImagesController.php
Comment (Progi1984) [reply controllers/admin/AdminImagesController.php]: Fixed

File: controllers/admin/AdminImagesController.php
Comment (Progi1984) [reply controllers/admin/AdminImagesController.php]: Fixed

**Comment by Progi1984** [comment]
@l-delin The wording is fixed :wink:

**Comment by Progi1984** [comment]
@zuk3975 Again sorry for the removal of your approval.

**Review by zuk3975 — DISMISSED** [review]

**Review by nicosomb — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]

**Comment by AureRita** [comment]
Hello @Progi1984 

Thank you for your PR, I test It and it doesn't seems to works, when I click on regenerate thumbnails, I have no modal. As you can see on this video : 

[Untitled_ Nov 17, 2022 4_50 PM.webm](https://user-images.githubusercontent.com/100699445/202497800-a0f0b147-708c-4376-a37d-af56c66bd2a9.webm)

Waiting for your feedback

**Comment by Progi1984** [comment]
@nicosomb @kpodemski @zuk3975 Hi, I just fixed the event on submit button (326da55) for @AureRita . Could you review it  when you have spare time ?

**Review by zuk3975 — APPROVED** [review]

**Review by marsaldev — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by sallemiines — COMMENTED** [review]
Hello @Progi1984 

Thanx for the PR !
It's LGTM QA :heavy_check_mark: you can check the screen record 

- Tests aut : **OK** 

**WITH PR**

[Untitled_ Nov 21, 2022 10_50 AM.webm](https://user-images.githubusercontent.com/31663369/203020108-01e84261-5569-49ee-8a0f-f6303248c999.webm)

**BEFORE PR :** 

[Untitled_ Nov 21, 2022 10_56 AM.webm](https://user-images.githubusercontent.com/31663369/203020808-6dbfa5b2-5b02-4f39-abe5-7db055945d2e.webm)

Thank you ! :rocket:

**Comment by marsaldev** [comment]
Thanks @Progi1984 !

**Pull request merged**
